wp-includes/sqlite/class-wp-sqlite-translator.php

```diff
@@ -1580,13 +1607,35 @@ private function execute_update() {
 
 			$this->rewriter->consume();
 		}
+
+		if ( $where && ( $limit || $order_by ) ) {
+			$this->rewriter->add( new WP_SQLite_Token( ')', WP_SQLite_Token::TYPE_OPERATOR ) );
+		}
+
 		$this->rewriter->consume_all();
 
 		$updated_query = $this->rewriter->get_updated_query();
 		$this->execute_sqlite_query( $updated_query, $params );
 		$this->set_result_from_affected_rows();
 	}
 
+	private function prepare_update_for_limit_or_order() {
+			$this->rewriter->add( new WP_SQLite_Token( ' ', WP_SQLite_Token::TYPE_WHITESPACE ) );
```

Comment:
I think there's a function to add many tokens at the same time. `add_many` perhaps?